ray-curator/ray_curator/stages/video/clipping/clip_extraction_stages.py

```diff
@@ -0,0 +1,386 @@
+import copy
+import pathlib
+import subprocess
+import uuid
+from dataclasses import dataclass
+from typing import Any
+
+from cosmos_xenna.ray_utils.resources import _get_local_gpu_info, _make_gpu_resources_from_gpu_name
+from loguru import logger
+
+from ray_curator.backends.base import WorkerMetadata
+from ray_curator.backends.experimental.ray_data.utils import RayStageSpecKeys
+from ray_curator.stages.base import ProcessingStage
+from ray_curator.stages.resources import Resources, _get_gpu_memory_gb
+from ray_curator.tasks import Clip, Video, VideoTask
+from ray_curator.utils import grouping
+from ray_curator.utils.operation_utils import make_pipeline_temporary_dir
+
+
+@dataclass
+class ClipTranscodingStage(ProcessingStage[VideoTask, VideoTask]):
+    """Stage that transcodes video clips into a standardized format.
+
+    This stage handles the conversion of video clips using FFmpeg, supporting both
+    software (libx264, libopenh264) and hardware (NVENC) encoding with configurable parameters.
+
+    Args:
+        num_cpus_per_worker: Number of CPUs per worker.
+        encoder: Video encoder to use.
+        encoder_threads: Number of threads per encoder.
+        encode_batch_size: Number of clips to encode in parallel.
+        nb_streams_per_gpu: Number of streams per GPU.
+        use_hwaccel: Whether to use hardware acceleration.
```

Comment:
It seems like even when `use_hwaccel` is False then we use `resources(gpus=1)` (at line 76). Can you elaborate what happens there?

Reply:
The else branch for L76 is for `if self.nb_streams_per_gpu > 0:` condition